### DISCUSSION TITLE
Using helloSystem to create music

### DISCUSSION THREAD

**Discussion by darkoverlordofdata** [helloSystem/hello#502] [Q&A]
Aside from the desktop aesthetics, one thing that drew me to helloSystem is the ‘Desktop system for creators…’ message. When I retired, I vowed that I would spend more time playing guitar, less time writing code.

But I’m not having much luck with this. Audacity is not a DAW, and LMMS doesn’t allow analog input. I recently found this article https://vermaden.wordpress.com/2023/03/02/freebsd-home-audio-studio/
I can’t get it to work (I tried on GhostBSD as well) but there is a glimmer of hope - starting with FreeBSD 13.1, I can assign realtime priority to individuals, and so jack2 doesn’t fail anymore.

Has anyone gotten something like Ardour working with Jack2 input? What did you have to do to get that to work?

Sigh - I’m still stuck on Ubuntu Studio until I can get this working on FreeBSD.

**Comment by darkoverlordofdata** [comment]
Progress: I now have something working, but it’s not pretty. There are 2 changes to vermaden’s instructions.

First, the ‘memorylocked’ issue. I changed the value in  /etc/login.conf
```
        :memorylocked=128M:\
```
And logged back in. That didn’t fix it. I rebooted. Still get the error. So I tried running as root (su), and no longer get this error. 

Second, the jack params:
 jackd_args="-r -d oss -r44100 -p1024 -n2 -w16 -i4 -o8 -C /dev/dsp0 -P /dev/dsp0"

Had to be changed to 
 jackd_args="-r -d oss -r44100 -p1024 -n2 -w16 -i4 -o8 -C /dev/dsp3 -P /dev/dsp3"

This will vary from system to system.You need to enter cat /dev/sndstat, and find your device:

pcm0: <Realtek ALC282 (Analog)> (play/rec)
pcm1: <Realtek ALC282 (Analog)> (play/rec)
pcm2: <Intel Panther Point (HDMI/DP 8ch)> (play)
pcm3: <USB audio> (play/rec) default

Mine is pcm3, a Behringer guitar/usb interface UC6102 from Sweetwater.

At this point, I can see progress - if I run jackd, qjackctl, and guitarix as root from the command line, it appears to work - I can see the sound visualizer going on the guitarix panel. But nothing makes it to the headphones. Typically, this means that I need to open the ALSA panel, and boost the recording level volume. But the OSS volume control doesn’t have a record channel volume slider. Not sure what’s up with that. I’ve noticed the overall output volume isn’t very loud, even at 100%, I can barely hear YouTube. So there are still some problems I have to dig into with the overall audio.

**Comment by probonopd** [comment]
Maybe it's just that my brain is incompatible to how jack works, but it never seems to "just work" for me. Always something needs to be configured first. Hence I am not going anywhere near it.

One setup for music on helloSystem that is working for me is described at
https://github.com/helloSystem/ISO/issues/261#issuecomment-1399275477
tl;dr: REAPER on WINE on helloSystem without JACK. The nice thing is that the majority of Windows-only VSTs can also work this way.

**Comment by darkoverlordofdata** [comment]
I’ve never used Reaper. I have been using Traktion Waveform, but they don’t have a FreeBSD version, so I was going to try Ardour, it’s well recommended. But so is Reaper.

In any case, my resolution for issue ‘Problems with audio Volume control #164’ seems to have fixed this as well. There are still some rough edges, but this works for me.

**Comment by darkoverlordofdata** [comment]
I have finally gotten this to work without root access. It was this memory locked parameter in /etc/login.conf:

```
:memorylocked=128M:\
```

I had to bump this way up to 1g
```
:memorylocked=1024M:\
```

That's a big chunk of memory; but I don't really use the computer for anything else while playing music, so from my POV this is solved.
